syft/execution/plan.py

```diff
@@ -384,6 +450,48 @@ def __setattr__(self, name, value):
             for tensor_name, tensor in value.named_tensors():
                 self.__setattr__(f"{name}_{tensor_name}", tensor)
 
+    def validate_input(self, args, kwargs):
+        """
+        Function used in __call__ to validate the input based on the input received at build time.
+
+        Args:
+            args: original args received on __call__.
+            kwargs: original kwargs received on __call__.
+        """
+
+        def extract_type(obj: PlaceHolder) -> str:
+            """
+            Function to extract type on the
+            """
+            for tag in obj.tags:
+                if "type_" in tag:
+                    return tag[6:]
+
+        inputs = sorted(self.find_placeholders("#fcall"), key=tag_sort("fcall"))
+
+        if len(inputs) != len(args):
+            raise RuntimeError(
+                f"Plan {self.name} requires {len(inputs)} arguments, received {len(args)}."
+            )
+
+        for index, arg in enumerate(args):
+            placeholder_type = extract_type(inputs[index])
+            argument_type = type(arg).__name__
+
+            tensor_type_error = (
+                isinstance(arg, FrameworkTensor) == placeholder_type == "FrameworkTensor"
```

Comment:
isinstance returns a bool, how could it be == to a "FrameworkTensor" str?

Reply:
My bad, this should work like a xor between isinstance(arg, FrameworkTensor) and  placeholder_type == "FrameworkTensor". Forgot a pair of parentheses, will fix in the next commit.